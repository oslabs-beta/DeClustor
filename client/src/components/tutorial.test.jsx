import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import Tutorial from './Tutorial';
import DashDemo from '../assets/dashboard.gif';

// Mock the image import
jest.mock('../assets/dashboard.gif', () => 'dashboard.gif');

describe('Tutorial Component', () => {
  it('renders correctly and matches the snapshot', () => {
    const { container } = render(<Tutorial />);
    expect(container).toMatchSnapshot();
  });

  it('renders the image with correct src and alt attributes', () => {
    const { getByAltText } = render(<Tutorial />);
    const imgElement = getByAltText('Tutorial');
    expect(imgElement).toBeInTheDocument();
    expect(imgElement).toHaveAttribute('src', 'dashboard.gif');
  });

  it('renders the Container and Box components with correct styles', () => {
    const { getByAltText, container } = render(<Tutorial />);
    const imgElement = getByAltText('Tutorial');
    const boxElement = imgElement.closest('div');
    expect(boxElement).toHaveStyle('display: flex');
    expect(boxElement).toHaveStyle('flexDirection: column');
    expect(boxElement).toHaveStyle('alignItems: center');
    expect(boxElement).toHaveStyle('marginTop: 8');
  });
});
