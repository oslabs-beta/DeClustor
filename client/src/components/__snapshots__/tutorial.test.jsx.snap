// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tutorial Component renders correctly and matches the snapshot 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd css-xdmu94-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-r5jd8k"
    >
      <img
        alt="Tutorial"
        height="auto"
        src="dashboard.gif"
        width="100%"
      />
    </div>
  </div>
</div>
`;
